oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/IndexUpdate.java

```diff
@@ -330,9 +334,14 @@ private void collectIndexEditors(NodeBuilder definitions,
                     // then we don't need to handle missing handler
                     if (definition.hasProperty(ASYNC_PROPERTY_NAME) && rootState.async == null) {
                         if (!TYPE_DISABLED.equals(type)) {
-                            log.warn("Missing provider for nrt/sync index: {} (rootState.async: {}). " +
-                                    "Please note, it means that index data should be trusted only after this index " +
-                                    "is processed in an async indexing cycle.", definition, rootState.async);
+                            long now = System.currentTimeMillis();
```

Comment:
`System.currentTimeMillis()` is not guaranteed to measure correctly the passage of time.

https://www.geeksforgeeks.org/java-system-nanotime-vs-system-currenttimemillis/ 
> It may give wrong results if the user changes the system time, hits a leap second or there are changes in NTP sync.

Even if in this use case it is not important if from time to time log messages are logged more than once a second or less frequently than once a second, I think we should be strictly correct, not to create confusion.

Reply:
I know System.currentTimeMillis() can go backwards, but in this are that's not a problem.

> not to create confusion.

The problem with nanotime is that it has so many zeroes, and that might be confusing as well... OK I can try.

Reply:
An alternative that relies in `System.nanoTime()` and is somewhat more readable is to use the StopWatch class from Guava. Just showing it here as a reference, not sure if it is worth to use it in this case:

```java
private static volatile Stopwatch lastMissingProviderMessageTime = Stopwatch.createStarted();


long sinceLastLogMillis = lastMissingProviderMessageTime.elapsed(TimeUnit.MILLISECONDS);
if (sinceLastLogMillis > 1000) {
   lastMissingProviderMessageTime = Stopwatch.createStarted();
```

Reply:
Probably we can do the same using the `Instant` class from the JDK, but internally Instant is more complex, as it tracks a point in real time, while StopWatch just tracks the passing of time without any anchor in real time.